cloudstack/zones.go

```diff
@@ -0,0 +1,84 @@
+package cloudstack
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+)
+
+// GetZone returns the Zone containing the current failure zone and locality region that the program is running in
+// In most cases, this method is called from the kubelet querying a local metadata service to aquire its zone.
+// For the case of external cloud providers, use GetZoneByProviderID or GetZoneByNodeName since GetZone
+// can no longer be called from the kubelets.
+func (cs *CSCloud) GetZone() (cloudprovider.Zone, error) {
+	return cloudprovider.Zone{}, nil
+}
+
+// GetZoneByProviderID returns the Zone containing the current zone and locality region of the node specified by providerId
+// This method is particularly used in the context of external cloud providers where node initialization must be down
```

Comment:
down => done
kubelets => kubelet

Reply:
Fixing, this was a copy/paste from the interface definition in kubernetes.

Reply:
Kept kubelets as it means all kubelets in all nodes.